utilities_test.go

```diff
@@ -96,7 +97,86 @@ func TestDatasetWarp(t *testing.T) {
 		t.Errorf("wrong block size %d,%d", st.BlockSizeX, st.BlockSizeY)
 	}
 }
+func TestDatasetWarpMulti(t *testing.T) {
+	ds1, _ := Open("testdata/test_1_1.tif", RasterOnly())
+	ds2, _ := Open("testdata/test_1_2.tif", RasterOnly())
+	ds3, _ := Open("testdata/test_2_1.tif", RasterOnly())
+	ds4, _ := Open("testdata/test_2_2.tif", RasterOnly())
 
+	defer ds1.Close()
+	defer ds2.Close()
+	defer ds3.Close()
+	defer ds4.Close()
+
+	// Warp NewDataset with multiple input dataset
+	filePath := path.Join(os.TempDir(), "warp.tif")
+	outputDataset, err := Warp(filePath, []*Dataset{ds1, ds2, ds3, ds4}, []string{"-ts", "40", "40"}, CreationOption("TILED=YES", "BLOCKXSIZE=32", "BLOCKYSIZE=16"), GTiff)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(filePath)
+	defer outputDataset.Close()
+
+	st := outputDataset.Structure()
+	if st.SizeX != 40 || st.SizeY != 40 {
+		t.Errorf("wrong size %d,%d", st.SizeX, st.SizeY)
+	}
+
+	if st.BlockSizeX != 32 || st.BlockSizeY != 16 {
+		t.Errorf("wrong block size %d,%d", st.BlockSizeX, st.BlockSizeY)
+	}
+
+	geoTransform, err := outputDataset.GeoTransform()
+	if err != nil {
+		t.Errorf("failed to get geotransform")
+	}
+
+	if geoTransform != [6]float64{45.000000, 0.250000, 0.000000, 35.000000, 0.000000, -0.250000} {
+		t.Errorf("wrong geotransform : [ %f, %f, %f, %f, %f, %f ]", geoTransform[0], geoTransform[1], geoTransform[2], geoTransform[3], geoTransform[4], geoTransform[5])
+	}
+}
+func TestDatasetWarpInto(t *testing.T) {
+	filePath := path.Join(os.TempDir(), "warpinto.tif")
+	outputDataset, err := Create(GTiff, filePath, 4, Byte, 10, 10)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(filePath)
+	defer outputDataset.Close()
+
+	sr, err := NewSpatialRefFromEPSG(4326)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = outputDataset.SetSpatialRef(sr); err != nil {
+		t.Fatal(err)
+	}
+	if err = outputDataset.SetGeoTransform([6]float64{45.000000, 1.0, 0.000000, 35.000000, 0.000000, -1.0}); err != nil {
+		t.Fatal(err)
+	}
+
+	inputDataset, _ := Open("testdata/test.tif", RasterOnly())
+	defer inputDataset.Close()
+
+	// Warp existing dataset with multiple input dataset
+	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-co", "TILED=YES"}); err == nil {
+		t.Error("All options related to creation ignored in update mode")
+	}
+
+	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-of", "GTiff"}); err == nil {
+		t.Error("All options related to creation ignored in update mode")
+	}
+
+	if err = outputDataset.WarpInto([]*Dataset{inputDataset}, []string{"-dstalpha"}, ConfigOption("GDAL_CACHEMAX=64")); err != nil {
+		t.Fatal(err)
+	}
+
+	if outputDataset.Structure().NBands != 4 {
+		t.Errorf("wrong band number : %d", outputDataset.Structure().NBands)
+	}
```

Comment:
Please add a test that shows that data in outputDataset has been updated with the contents of inputDataset